test/src/test/java/org/corfudb/runtime/object/transactions/StreamTest.java

```diff
@@ -289,4 +290,20 @@ public void testConcurrentStreamRW() {
     }
 
 
+    @Test (expected = TransactionAbortedException.class)
+    public void checkTransactionAbortIfLookupByIndexWhenIndexNotSpecified() {
+        CorfuTable<String, String> map = instantiateCorfuObject(CorfuTable.class, "Test");
+
+        try {
+            TXBegin();
+            map.put("ak", "av");
+            map.getByIndex(StringIndexer.BY_FIRST_LETTER, "a");
+            TXEnd();
+        } catch (TransactionAbortedException tae) {
+            assertThat(tae.getAbortCause()).isEqualTo(AbortCause.UNDEFINED);
+            assertThat(tae.getCause().equals(IllegalArgumentException.class));
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Complete the assertion. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2970)